core/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -220,14 +247,17 @@ private[spark] class SparkConcurrentModificationException(
 private[spark] class SparkDateTimeException(
     errorClass: String,
     errorSubClass: Option[String] = None,
-    messageParameters: Array[String],
+    messageParameters: Map[String, String],
```

Comment:
This is tested at least by QueryExecutionAnsiErrorsSuite "CANNOT_PARSE_TIMESTAMP: parse string to timestamp"